### ISSUE TITLE
Warn user if cannot write PP3

### ISSUE THREAD

**Issue by Beep6581** [Beep6581/RawTherapee#4212]
Preferences > Image Processing > Processing profile saving priority
When that is set to save next to the input file but saving there is no possible, e.g. because the destination is read-only, RT currently does not warn the user, and so the processing settings will be lost when closing the image.

To test:
```
mkdir /tmp/ro
cp ~/photos/test_images/amsterdam.pef /tmp/ro
chmod 555 /tmp/ro
```

**Comment by heckflosse** [comment]
@Beep6581 In a first step we could use your wording
```
Preferences > Image Processing > Processing profile saving priority
When that is set to save next to the input file but saving there is no possible, e.g. because the destination is read-only, RT currently does not warn the user, and so the processing settings will be lost when closing the image.
```
as a tooltip in preferences ;-)

**Comment by Beep6581** [comment]
Yes, that would be a very unexpected (to me) workaround to this issue ;)

**Comment by Beep6581** [comment]
Related #1688

**Comment by tomato42** [comment]
I've just lost like 6h of work because of this "feature"

Saving the pp data into some arbitrary folder (_not_ shared between AppImage and native version) is *very* confusing and *very* unexpected application behaviour.

Please, change it so that presents a clear and obvious error message when it cannot save the profile next to the input file!

**Comment by Lawrence37** [comment]
@tomato42

Can you specify what "feature" you are referring to?

Which "arbitrary folder" are you talking about? Is it the cache? Please note that we only distribute AppImages for Linux, so we don't have absolute control over other packages.

**Comment by tomato42** [comment]
the feature of writing the profile to any directory but the one that has the edited raw file

**Comment by Lawrence37** [comment]
The feature is configurable in Preferences and [documented in the manual](https://rawpedia.rawtherapee.com/Preferences#Processing_Profile_Handling). By default, it is set to save pp3s next to the image. The cache location is [well-defined](https://rawpedia.rawtherapee.com/File_Paths#Cache).

Is RawTherapee attempting and failing to write pp3s next to the source image, or do you have it configured to save to the cache only? If it's saving to the cache, you can access your edits if the processing profile loading priority is set to the cache.

**Comment by tomato42** [comment]
Just because something is documented in the manual doesn't mean that it's the behaviour users expect of the application.

It failed to write because I mounted the location with the raw images with wrong settings so my user didn't have write permission. I only noticed because I started to export the files and it didn't create the `converted` subdirectory, _again_ without raising any errors.

Maybe I can change the settings so that it uses the cache instead of the sidecar files, it didn't work for me, the edits weren't there any more.

What I did:
1. have files in a directory, do basic editing, leave it
2. mount it later, such that the user has no write permission (through regular unix permissions)
3. edit the files some more
4. remount with correct permissions
5. open the directory - all edits are gone
6. changing the settings so that it prioritises the cache dir
7. still no second editing visible

**Comment by Lawrence37** [comment]
> Just because something is documented in the manual doesn't mean that it's the behaviour users expect of the application.

I disagree for various reasons, but that's a topic for a separate discussion. It's not relevant anymore to this issue since we now know the program was configured to write pp3s in the same directory as your images.
